src/Tools/babylon.assetsManager.ts

```diff
@@ -669,6 +669,118 @@ module BABYLON {
+        public onSuccess: (task: ImageAssetTask) => void;
+        /**
+         * Callback called when the task is successful
+         */
+        public onError: (task: ImageAssetTask, message?: string, exception?: any) => void;
+
+        /**
+         * Creates a new AtlasTextureAssetTask
+         * @param name defines the name of the task
+         * @param url defines the location of the atlas texture to load
+         */
+        constructor(public name: string, public url: string) {
+            super(name);
+        }
+
+        /**
+         * Execute the current task
+         * @param scene defines the scene where you want your assets to be loaded
+         * @param onSuccess is a callback called when the task is successfully executed
+         * @param onError is a callback called if an error occurs
+         */
+        public runTask(scene: Scene, onSuccess: () => void, onError: (message?: string, exception?: any) => void) {
+            let loadJsonFile = () => {
+                scene._loadFile(this.url, (data) => {
+                    let jsonData = JSON.parse(data as string);
+                    if (jsonData.meta) {
+                        this._framesData = jsonData.frames;
+                        let imageName = jsonData.meta.image;
+                        let parentPath = Tools.GetFolderPath(this.url);
+                        loadImageFile(parentPath + imageName);
+                    } else
+                        Tools.Error('Not found meta tag in JSON file: ' + this.url);
+                }, undefined, false, false, (request, exception) => {
+                    if (request) {
+                        onError(request.status + " " + request.statusText, exception);
+                    }
+                });
+            };
+
+            let loadImageFile = (url: string) => {
+                Tools.LoadImage(url, (img) => {
+                    this._parseData(img);
+                    onSuccess();
+                }, function (message, error) {
+                    onError(message, error);
+                }, null);
+            };
+
+            loadJsonFile();
+        }
+
+        /**
+         * Data parsed when load success JSON and Image file
+         * @param {HTMLImageElement} img
+         * @private
+         */
+        private _parseData(img: HTMLImageElement) {
+            Object.keys(this._framesData).forEach((key) => {
+                let name = key.substring(0, key.lastIndexOf('.'));
+                this.data[name] = new AtlasTexture(name, img, this._framesData[key]);
+            });
+        }
+    }
+
+    /**
+     * Defines the interface used storage information atlas texture
+     */
+    export interface IAtlasTexture {
+        frame: any,
+        pivot: any,
+        rotated: boolean,
+        sourceSize: any,
+        spriteSourceSize: any,
+        trimmed: boolean,
+    }
+
+    /**
+     * The class used for source's GUI.Image
+     */
+    export class AtlasTexture {

```

Comment:
This is not a texture but more a Rect with an imageElement? (Not a texture as in Babylon.Texture I mean)

Reply:
As it is designed to work with GUI.Image we could name it GUI.ImageAtlas ?

Reply:
Yes, this is not a texture. Your means is create new class ImageAtlas extend Image?

Reply:
Yep just anew ImageAtlas only

Reply:
After thinking again, i see no necessary to create a new class ImageAtlas.
1) Use the class Image, it will be flexible when change src. Example: user get avatar (string) from server, for some reason the path does not exist, developer are easy fallback to load avatar default from AtlasTexture.
2) Actually i have created and used AtlasTexture in my project 3 months ago. So far, it still working at well, no problem.

Reply:
Well please update the PR to see how it looks like:)